public/app/plugins/datasource/elasticsearch/specs/elastic_response.test.ts

```diff
@@ -704,17 +841,24 @@ describe('ElasticResponse', () => {
       };
 
       result = new ElasticResponse(targets, response).getTimeSeries();
-    });
+      expect(result.data.length).toBe(1);
 
-    it('should include field in metric name', () => {
-      expect(result.data[0].type).toBe('table');
-      expect(result.data[0].rows[0][1]).toBe(1000);
-      expect(result.data[0].rows[0][2]).toBe(3000);
+      expect(result.data[0].length).toBe(1);
+      expect(result.data[0].fields.length).toBe(3);
+      expect(result.data[0].fields[0].name).toBe('host');
+      expect(result.data[0].fields[0].values.length).toBe(1);
+      expect(result.data[0].fields[1].name).toBe('Average test');
+      expect(result.data[0].fields[1].values.length).toBe(1);
+      expect(result.data[0].fields[2].name).toBe('Average test2');
+      expect(result.data[0].fields[2].values.length).toBe(1);
+      expect(result.data[0].fields[0].values.get(0)).toBe('server-1');
+      expect(result.data[0].fields[1].values.get(0)).toBe(1000);
+      expect(result.data[0].fields[2].values.get(0)).toBe(3000);
```

Comment:
if you use DataFrameView you can get an object for each row (with a property for each field in frame)

Reply:
Thanks for this tip. Don't know how I didn't notice it while looking at this file as it was even used in one of the tests. Very useful.